src/Kiota.Builder/Refiners/JavaRefiner.cs

```diff
@@ -18,6 +18,14 @@ public override Task Refine(CodeNamespace generatedCode, CancellationToken cance
         {
             cancellationToken.ThrowIfCancellationRequested();
             LowerCaseNamespaceNames(generatedCode);
+            var reservedNamesProvider = new JavaReservedNamesProvider();
+            CorrectClassNames(generatedCode, s =>
+            {
+                if (!reservedNamesProvider.ReservedNames.Contains(s) && s.Contains('_'))
```

Comment:
I believe it is possible for the result of `s.ToPascalCase(new[] { '_' })` be in the `ReservedNames` so we should check for that too. (e.g an original class name of `stack_overflow_exception` would be returned as `StackOverflowException` even if `StackOverflowException` is in the reserved words.

Reply:
done